app/controllers/courses_controller.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class CoursesController < ApplicationController
+  def index
+    if current_user.student?
+      @courses = Course.groups_for_user(current_user).includes(:subject)
```

Comment:
You can use`current_user.group.courses` but must add `belongs_to :group` in User model